LuckyDragon/app/src/main/java/com/example/luckydragon/Models/UserList.java

```diff
@@ -41,8 +39,8 @@ public UserList(FirebaseFirestore db) {
             }
             if (value != null) {
                 users.clear();
-                for (QueryDocumentSnapshot doc: value) {
-                    users.add(createUser(doc));
+                for(int i = 0; i < value.size(); i++) {
+                    users.add(createUser((QueryDocumentSnapshot) value.getDocuments().get(i)));
```

Comment:
Any reason why this loop is changed?

Reply:
It is easier to mock a basic for loop than one that uses iterators.